api/auth/auth-middleware.js

```diff
@@ -63,7 +103,27 @@ const validateRoleName = (req, res, next) => {
       "message": "Role name can not be longer than 32 chars"
     }
   */
-  next();
+const validateRoleName = async (req, res, next) => {
+  try {
+    const { role_name } = req.body;
```

Comment:
daca role_name vine ca "admin", in care `if` intra el?

Reply:
Vine in primul if, in general orice rol as da el va nimeri in primul if mereu pentru ca va fi true. Asta e si problema de fapt, nu inteleg ce fel de validare trebuie sa fie in primul if. Nu vad sens in aceasta conditie _" If the role_name in the body is valid"_ , ce fel de validare se cere?

Reply:
Ideea e foarte simpla. Mai intai verifici daca e valid, apoi in tot acelasi if verifici deja daca rolul e admin si cealalta verificare daca e mai lung de 32 de caractere. Deci o sa ai 2 if-uri in interiorul la alt if

Reply:
Bine, m-am gandit la ast ade fapt. Si inca un if pentru cazul acesta **if (!role_name || trimmedRoleName === "")** ??